IotHubConsumer/ReceiverConsumer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+using Microsoft.Extensions.Configuration;
+
+namespace IotHubConsumer
+{
+    public class ReceiverConsumer
+    {
+        private readonly IConfiguration configuration;
+        public HashSet<TestElement> MessagesWithRetainSet { get; set; } = new HashSet<TestElement>();
```

Comment:
Why did decide to use a hashset? :)

Reply:
It started as a hashset of strings for which I wanted to have retrieval in O(1) when searching for a value. Depending on whether we decide to keep the TestElement class or not, I could also create a Dictionary with the guid of the message as key, but that might complicate adding the other properties as value.
Which other data structures did you have in mind?

Reply:
I'm pretty simple minded when it comes to data types and would go with icollectible of type partitionEvent. I made the suggestion of partitionEvent, because you're currently sending two parameters (plus more to come for other tests) from the event anyway, so why not just send the entire event?

I asked about the hash because I was curious :)

Reply:
@Ioana37 Did you use `TestElement` to avoid deserialization which would happen in case of `partitionEvent`?